src/kernel/include/shared/cos_types.h

```diff
@@ -287,11 +311,27 @@ enum
 	BOOT_MEM_KM_BASE = PGD_SIZE, /* kernel & user memory @ 4M, pgd aligned start address */
 };
 
-enum
-{
+typedef enum {
 	/* thread id */
 	THD_GET_TID,
-};
+	/* get regs */
+	THD_GET_FAULT_REG0,
```

Comment:
Are fault regs different from regular regs? Even if they represent a "fault" context and not regular context, I think the registers are still same. My point is, `FAULT_IP == THD.FAULT.IP` in some sense. Perhaps instead of having different set of enums here, have some kind of `fault_introspect` that passes an in_fault flag and fetches fault_regs?

Reply:
Also I don't think we should have `REGX` notation for `IP` and `SP` at the very least?